api/workers.go

```diff
@@ -36,6 +45,15 @@ func (a *API) workerGetJob(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Check if worker is available
+	if available, err := a.jobsManager.isWorkerAvailable(workerAddress); !available {
```

Comment:
This system has a vulnerability. It is very easy to impersonate a worker (just by knowing the address used, which is public available). Not something to fix now but we must at least open an issue.